### PR TITLE
Include islandora_fits into islandora/modules

### PR DESCRIPTION
**Pull request by rosiel**
**GitHub Issue**: https://github.com/Islandora/islandora/issues/906

# What does this Pull Request do?

Includes Islandora Fits as a submodule of Islandora.

# What's new?

* Islandora Fits submodule, including its git history.
* Added a dependency to `composer.json` for ext-simpleXML, required by Islandora FITS so it can parse XML.

* Does this change add any new dependencies? - yes, simpleXML. will be installed automatically when using Composer to update islandora.
* Does this change require any other modifications to be made to the repository
 (i.e. Regeneration activity, etc.)? 
* Could this change impact execution of existing code? no, shouldn't.

# How should this be tested?

To upgrade to this version:
* add this branch as a repository in your composer.json, and change the spec for "islandora/islandora" to "dev-include-fits" (won't be required after merging).
* `composer remove "islandora-rdm/islandora_fits"`
* `composer update -W "islandora/islandora"` (make sure you get the new code)
* cache refresh and restart your webserver so PHP finds the files in their new locations.
* FITS derivatives should work exactly as before.
* Note: I did NOT merge the `islandora_fits_tags` migration with the islandora tags migration. I'm not sure we should, since this is an optional module. But i'm open to suggestions.

# Documentation Status

* Does this change existing behaviour that's currently documented? - no, same behaviour
* Does this change require new pages or sections of documentation? - no
* Who does this need to be documented for? - n/a
* Associated documentation pull request(s): ___  or documentation issue ___

# Deployment Status (post-commit)

* Starter Site - will need an update (removing islandora_fits from composer when locking to this version of islandora)
* Install Profile - will need an update (removing islandora_fits)

# Additional Notes:
Any additional information that you think would be helpful when reviewing this
 PR.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/committers

### CODE REVIEW

**Comment by adam-vessey** [comment]
Don't appear to be able to comment on it directly, but `modules/islandora_fits/migrations/.swp` probably shouldn't exist.

File: modules/islandora_fits/README.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+Restart Tomcat and test with:
+```bash
+curl -k -F datafile="@/path/to/myfile.jpg" http://example.com:8080/fits/examine
+```
+(note: the ‘@’ is required.)
+
+#### Installing Microservice
+Get code from https://github.com/roblib/CrayFits and install.  This code can live anywhere, including an external server,
```
Comment (adam-vessey) [inline modules/islandora_fits/README.md]: Making reference to the roblib org specifically; should be updated to point at however it gets moved over to the IF, whenever it happens (not sure if there's an issue/PR for it, yet?).

File: modules/islandora_fits/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+On a production machine you'd probably want to configure an additional
+port in Apache.
+
+Note: The location of the FITS webserver is stored in the `.env` file in the
+root dir of the Symfony app.  This will have to be reconfigured if the FITS
+server is anywhere other than `localhost:8080/fits`
+
+#### Adding FITs requests to the queue
+Copy the file `assets/ca.islandora.alpaca.connector.ocr.blueprint.xml`
+to `/opt/karak/deploy` on your server.  There is no need to restart.
```
Comment (adam-vessey) [inline modules/islandora_fits/README.md]: Making reference to Karaf here (assuming? `karak`?); however, Karaf is no longer a required component in newer Alpaca.

Does seem to be something of a suggestion of what to do with FITS, in the [Alpaca README](https://github.com/Islandora/Alpaca/blob/2.x/README.md), with:

```
derivative.fits.enabled=true
derivative.fits.in.stream=queue:islandora-connector-fits
derivative.fits.service.url=http://127.0.0.1:8000/crayfits
derivative.fits.concurrent-consumers=2
derivative.fits.max-concurrent-consumers=2
derivative.fits.async-consumer=false
```

File: modules/islandora_fits/README.md (hunk 1 above)
Comment (rosiel) [reply modules/islandora_fits/README.md]: there's an [issue](https://github.com/Islandora/Crayfish/issues/173) but no PR. I saw there's currently[ work being done for PHP 8.1](https://github.com/roblib/CrayFits/pull/10) and figured it'd be good to wait till that's in.

Comment (adam-vessey) [reply modules/islandora_fits/README.md]: Ah. I had looked in `islandora/documentation` for the issue.

File: modules/islandora_fits/README.md (hunk 2 above)
Comment (rosiel) [reply modules/islandora_fits/README.md]: I moved the general "configure the microservice" stuff into the CrayFits readme ([new PR](https://github.com/roblib/CrayFits/pull/11/files)). I don't think it really belongs in the drupal submodule. 

However this module does install configuration, which makes assumptions about how your alpaca is configured. I added more in the "requirements" section of this readme, namely:

> - An [Alpaca](https://github.com/Islandora/Alpaca/tree/2.x/islandora-connector-derivative) `islandora-connector-derivative` configured for CrayFits. The default configuration in this module assumes the queue is named `islandora-connector-fits`.

**Comment by rosiel** [comment]
Closing as the composer requirements exceed what you might need with an islandora. Instead, I'll see about transferring the repo to the Foundation.

**Pull request closed without merging**